planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -2651,7 +2658,39 @@ TurnSignalInfo AvoidanceModule::calcTurnSignalInfo(const ShiftedPath & path) con
       turn_signal_info.turn_signal.command = TurnIndicatorsCommand::ENABLE_RIGHT;
     }
   } else {
-    turn_signal_info.turn_signal.command = TurnIndicatorsCommand::DISABLE;
+    // Create vehicle footprint
+    const auto local_vehicle_footprint =
+      createVehicleFootprint(planner_data_->parameters.vehicle_info);
+    const auto vehicle_footprint = transformVector(
+      local_vehicle_footprint,
+      tier4_autoware_utils::pose2transform(planner_data_->self_odometry->pose.pose));
+    for (const auto & cl : avoidance_data_.current_lanelets) {
+      if (segment_shift_length > 0.0) {
+        // Get left bounds
+        const auto left_bound = cl.leftBound();
+        // convert to boost polygon for intersection check
+        Polygon2d left_bound_polygon;
+        for (const auto & p : left_bound) {
+          left_bound_polygon.outer().push_back(Point2d(p.x(), p.y()));
+        }
+        if (boost::geometry::intersects(left_bound_polygon, vehicle_footprint)) {
```

Comment:
You shouldn't convert a linestring to a polygon if it is not needed so. Naturally a lanelet bound is a linestring so we should use it that way. Using as polygon is not right plus it is more costly and more complex. You can do it as below and please apply the same for right bound :

```suggestion
        const auto left_bound = cl.leftBound2d().basicLineString();
        if (boost::geometry::intersects(left_bound, vehicle_footprint)) {
```